File: jsonbench.rb
#!/usr/bin/ruby

require 'json'

def main
	repeat   = ARGV[0].to_i
	filename = ARGV[1]
	buf      = File.read(filename)
	duration = 0

	for _ in 0...repeat
		start = Time.now
		JSON.parse(buf)
		duration += Time.now - start
	end

	puts (duration * 1000).to_i
end

main if __FILE__ == $0

File: jsonbench.py
#!/usr/bin/python

from __future__ import division, print_function, with_statement

from time import clock
from json import loads

def main(args):
	repeat, filename = args
	repeat = int(repeat)
	with open(filename, "rb") as f:
		buf = f.read().decode('utf-8')

	duration = 0
	for _ in range(repeat):
		start = clock()
		loads(buf)
		duration += clock() - start
	print(int(duration * 1000))

if __name__ == '__main__':
	import sys
	main(sys.argv[1:])
